File: app/src/main/java/com/mesada/shop/repository/CategoryRepository.java
package com.mesada.shop.repository;

import android.content.Context;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.mesada.shop.model.Product;
import com.mesada.shop.model.ProductCategory;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;
import java.io.InputStream;
import java.util.ArrayList;
import java.util.List;

public class CategoryRepository {

    private MutableLiveData<List<ProductCategory>> mutableProductCategory;

    public LiveData<List<ProductCategory>> getProductCategory() {
        if (mutableProductCategory == null) {
            mutableProductCategory = new MutableLiveData<>();
            loadProductsCategory();
        }
        return mutableProductCategory;
    }

    private void loadProductsCategory() {
        List<ProductCategory> productCategoryList = new ArrayList<>();
        productCategoryList.add(new ProductCategory(1, "All"));
        productCategoryList.add(new ProductCategory(2, "Jacket"));
        productCategoryList.add(new ProductCategory(3, "Blazer"));
        productCategoryList.add(new ProductCategory(4, "Tee"));

        mutableProductCategory.setValue(productCategoryList);
    }
}

File: app/src/main/java/com/mesada/shop/repository/CategoryRepository.kt
package com.mesada.shop.repository

import androidx.lifecycle.MutableLiveData
import com.mesada.shop.model.ProductCategory
import androidx.lifecycle.LiveData
import com.mesada.shop.model.Cart
import java.util.*

class CategoryRepository {
    private var mutableProductCategory: MutableLiveData<List<ProductCategory>>? = null
    private val mutableCategory = MutableLiveData<String?>()

    val productCategory: LiveData<List<ProductCategory>>
        get() {
            if (mutableProductCategory == null) {
                mutableProductCategory = MutableLiveData()
                loadProductsCategory()
            }
            return mutableProductCategory!!
        }

    private fun loadProductsCategory() {
        val productCategoryList: MutableList<ProductCategory> = ArrayList()
        productCategoryList.add(ProductCategory(1, "All"))
        productCategoryList.add(ProductCategory(2, "Jacket"))
        productCategoryList.add(ProductCategory(3, "Blazer"))
        productCategoryList.add(ProductCategory(4, "Tee"))
        mutableProductCategory!!.value = productCategoryList
    }

    val SelectedCategory: LiveData<String?>
        get() {
            if (mutableCategory.value == null) {
                mutableCategory.value = "all"
            }
            return mutableCategory
        }

    fun selectCategory(category: String?) {
        if (mutableCategory.value == null) return
        mutableCategory.value = category
    }
}